HW1/P2/P2b.py

```diff
@@ -1,3 +1,40 @@
 from P2 import *
+from pyspark import SparkContext
+import numpy as np
+import matplotlib.pyplot as plt
+from random import randint
 
-# Your code here
+# Convert pixel (i, j) to the correct (x, y) value
+def pixelize(x):
+    return x / 500. - 2
+
+# Number of x and y pixels
+NX = NY = 2000
+
+if __name__ == "__main__":
+
+    # Declare our SparkContext
+    sc = SparkContext("local", "madelbrot")
+
+    # Get the i, j pixel arrays
+    xs = sc.parallelize(range(0, NX), 10)
+    ys = sc.parallelize(range(0, NY), 10)
+
+    # Take the cartesian product to get the pixel grid
+    # Now we partition into 100 partitions, and to determine where each element
+    # of the RDD goes, we simply assign a random number between 1 and 100 (inclusive)
+    pixels = xs.cartesian(ys).partitionBy(100, lambda p: randint(1, 101) )
+
```

Comment:
Nice solution.  You might also consider [`.repartition(numPartitions)`](https://spark.apache.org/docs/1.2.0/programming-guide.html), which randomly assigns elements to `N = numPartitions` partitions.  Also, careful with `randint` - your upper bound here is 101, which is one too many.  You used the [`random` module's version](https://docs.python.org/2/library/random.html#random.randint
) of this function, which is upper bound inclusive, so your range includes 101.  ([Numpy's randint](http://docs.scipy.org/doc/numpy/reference/generated/numpy.random.randint.html), on the other hand, is upper bound exclusive.)